examples/face_recognition_11/face_recognition_11/utils.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+from typing import List
+from typing import Tuple
+
+import numpy as np
+from face_recognition_11.workflow import GroundTruth
+from face_recognition_11.workflow import Inference
+from face_recognition_11.workflow import TestSample
+from face_recognition_11.workflow import TestSampleMetrics
+
+from kolena.workflow import Histogram
+
+
+def compute_threshold(
+    test_samples: List[TestSample],
+    ground_truths: List[GroundTruth],
+    inferences: List[Inference],
+    fmr: float,
+    eps: float = 1e-9,
+) -> float:
+    # address duplicates
```

Comment:
Isn't threshold computed only for the complete test case? If so, can't we just use the ` / 2` trick here?

Reply:
If we want to use this logic, `seen` should be a dict not a list.  Checking through a list should be O(n) and it becomes too expensive.